### PR TITLE
feat: allow `core` to fetch credentials from `oauth`

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

We need `core` to be able to get a `crendential` content from `oauth`.

To achieve this:

// Done with this PR:
- whitelist `core` -> `oauth` requests in network policy 

// Already done manually:
- add an API for `core` to talk to `oauth` in `oauth` API keys file
- add an env var for the `OAUTH_API_KEY` on `core`
- add an env var for the `OAUTH_API` on `core`

also added a quick client to fetch the credential content on `oauth` from `core`

## Risk

N/A

## Deploy Plan

- apply infra
- deploy `core` and `oauth`

### CODE REVIEW

File: core/src/oauth/client.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            .header("Content-Type", "application/json")
+            .header("Authorization", format!("Bearer {}", *OAUTH_API_KEY))
+            .send()
+            .await?;
+
+        match res.status().as_u16() {
+            200 => {
+                let body = res.text().await?;
+                let json = serde_json::from_str::<serde_json::Value>(&body)?;
+                let content = json["content"].to_string();
```
Comment (Fraggle) [inline core/src/oauth/client.rs:25]: You don't want a serde_json:Map:<.... ?

**Review by Fraggle — APPROVED** [review]

File: core/src/oauth/client.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/oauth/client.rs:28]: No because I want to parse into a specific struct depending on which provider it is

Comment (Fraggle) [reply core/src/oauth/client.rs:28]: I think, you can use serde_json::from_value().

Comment (fontanierh) [reply core/src/oauth/client.rs:28]: fair 👍

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**